auto_changelog/repository.py

```diff
@@ -135,20 +134,29 @@ def _extract_note_args(commit) -> Tuple[str, str, str, str, str, str]:
         """ Extracts arguments for release Note from commit """
         sha = commit.hexsha
         message = commit.message
-        type_, scope, description, body, footer = GitRepository._parse_conventional_commit(message)
-        return sha, type_, description, scope, body, footer
+        type_, scope, description, body, footer, breaking_change = GitRepository._parse_conventional_commit(message)
+        return sha, type_, description, scope, body, footer, breaking_change
 
     @staticmethod
     def _parse_conventional_commit(message: str) -> Tuple[str, str, str, str, str]:
-        type_ = scope = description = body = footer = ""
+        type_ = scope = breaking_flag = breaking_change = description = body = footer = ""
         # TODO this is less restrictive version of re. I have somewhere more restrictive one, maybe as option?
-        match = re.match(r"^(\w+)(\(\w+\))?: (.*)(\n\n.+)?(\n\n.+)?$", message)
+        match = re.match(r"^(\w+)(\(\w+\))?(!)?: (.*)(\n\n.+)?(\n\n.+)?$", message)
         if match:
-            type_, scope, description, body, footer = match.groups(default="")
+            type_, scope, breaking_flag, description, body, footer = match.groups(default="")
         if scope:
             scope = scope[1:-1]
+        if breaking_flag:
+            breaking_change = description
         if body:
             body = body[2:]
+            if body.strip().startswith("BREAKING CHANGE: "):
```

Comment:
Hello ! 

This is cool addition ! I looked at the change seems good for a first version of the support of breaking changes.
Just wanted to point out that this is not allowed that `BREAKING CHANGE:`  in the body. Breaking changes **MUST** be indicated in the type/scope prefix of a commit, or as an entry in the footer. So this part linked to the body is not necessary.

Reply:
Thank for review.

I admit this line was a dirty hack to handle messages without body...

I reworked commit message parsing to correctely extract footers :
> One or more footers MAY be provided one blank line after the body. Each footer MUST consist of a word token, followed by either a :<space> or <space># separator, followed by a string value (this is inspired by the git trailer convention).

Reply:
Ok, I read it.. and have few questions. From points 8 and 9 of the specification:

> 8. Breaking changes MUST be indicated at the very beginning of the footer or body section of a commit. A breaking change MUST consist of the uppercase text BREAKING CHANGE, followed by a colon and a space.
> 9. A description MUST be provided after the BREAKING CHANGE:, describing what has changed about the API, e.g., BREAKING CHANGE: environment variables now take precedence over config files.

So why did you choose so complicated approach?
What you can do is simply match the beginning of the body/footer. Then parse breaking change description without removing it from the sections. It's still part of them. So for example:
```python
if body:
    body = body[2:]
    breaking_change = extract_breaking_change(body)  # returns breaking description
if footer:
    footer = footer[2:]
    breaking_change = extract_breaking_change(footer)
```
Now we have a few points for discussion:
- How will we deal with cases when both, body and footer will contain BREAKING CHANGE?
- How will we deal with cases when multiple BREAKING CHANGE comments will be in one section?

Note: If we keep breaking change in original places, we may move extraction of them into domain. 

I like how you did the domain logic and presentation layer.